pkg/policy/policies.go

```diff
@@ -44,127 +81,83 @@ type Policies struct {
 	containerFiltersEnabled   uint64 // bitmap of policies that have at least one container filter type enabled
 }
 
-func NewPolicies() *Policies {
+func NewPolicies(cfg PoliciesConfig) *Policies {
 	return &Policies{
 		rwmu:                      sync.RWMutex{},
+		config:                    cfg,
 		version:                   0,
 		bpfInnerMaps:              map[string]*bpf.BPFMapLow{},
-		policiesArray:             [MaxPolicies]*Policy{},
+		policiesArray:             [PolicyMax]*Policy{},
 		filterEnabledPoliciesMap:  map[*Policy]int{},
+		evtsStates:                newEventsStates(),
 		filterUserlandPoliciesMap: map[*Policy]int{},
 		uidFilterMin:              filters.MinNotSetUInt,
 		uidFilterMax:              filters.MaxNotSetUInt,
 		pidFilterMin:              filters.MinNotSetUInt,
 		pidFilterMax:              filters.MaxNotSetUInt,
 		uidFilterableInUserland:   false,
 		pidFilterableInUserland:   false,
-		filterableInUserland:      0,
-		containerFiltersEnabled:   0,
+		filterableInUserland:      PolicyNone,
+		containerFiltersEnabled:   PolicyNone,
 	}
 }
 
-func (ps *Policies) Count() int {
+// EventsStates returns the events states of Policies.
+func (ps *Policies) EventsStates() events.EventsStates {
 	ps.rwmu.RLock()
 	defer ps.rwmu.RUnlock()
 
-	return len(ps.filterEnabledPoliciesMap)
+	return ps.evtsStates
```

Comment:
For all thread-safe logic, let's assume what I mentioned in https://github.com/aquasecurity/tracee/pull/3848#issuecomment-1969162026 - _Policies probably won't require locking anymore, however, we're sticking with it until have moved that control to PolicyManager._

That's because that reference can be stale:

https://github.com/aquasecurity/tracee/pull/3848/files#diff-4c934aefc1504cfe031f05bbeec5de2c17ffb211106c1368771871b92284a8f7R89